src/cursor/common.rs

```diff
@@ -37,8 +38,10 @@ where
     client: Client,
     info: CursorInformation,
     buffer: VecDeque<RawDocumentBuf>,
+    post_batch_resume_token: Option<ResumeToken>,
```

Comment:
Two tokens need to be tracked - the one returned by the most recent `getMore` call, and the one that will be returned to users via `resume_token()`.  The two are only the same at the end of a batch.